api/events/src/main/java/io/opentelemetry/api/events/EventEmitterProvider.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.events;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A registry for creating scoped {@link EventEmitter}s. The name <i>Provider</i> is for consistency
+ * with other languages and it is <b>NOT</b> loaded using reflection.
+ *
+ * @see EventEmitter
+ */
+@ThreadSafe
+public interface EventEmitterProvider {
+
+  /**
+   * Gets or creates a named EventEmitter instance which emits events to the {@code eventDomain}.
+   *
+   * @param instrumentationScopeName A name uniquely identifying the instrumentation scope, such as
+   *     the instrumentation library, package, or fully qualified class name. Must not be null.
+   * @param eventDomain The event domain, which acts as a namespace for event names. Within a
+   *     particular event domain, event name defines a particular class or type of event.
+   * @return a Logger instance.
+   */
+  default EventEmitter get(String instrumentationScopeName, String eventDomain) {
```

Comment:
Is eventDomain mandatory? I don't even know what sort of values I'd put in for this.

Reply:
As of now `event.domain` is [required](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/semantic_conventions/events.md). The spec describes it as:
> The event.domain attribute is used to logically separate events from different systems. For example, to record Events from browser apps, mobile apps and Kubernetes, we could use browser, device and k8s as the domain for their Events. This provides a clean separation of semantics for events in each of the domains. Within a particular domain, the event.name attribute identifies the event. Events with same domain and name are structurally similar to one another.

There's an open [issue](https://github.com/open-telemetry/opentelemetry-specification/issues/2994) to discuss whether event.domain and event.name can be merged into a single field.

Reply:
This is the only sticking point I have with this API. I don't love having to explain to people what this is supposed to be used for. We have had enough trouble explaining what the "name" of a Tracer is...this adds an additional layer of having to explain this arcane idea that almost no one will care about. Can we make it optional? Or default to something like "none" or "unknown" or "who cares?"? :)

Reply:
I'm not opposed to making it optional with a default given the experimental state of the spec. The spec could reaffirm the importance and requirement of this field, in which case we'd have to come back and adjust our API, but that's fine and won't be the last breaking change to the event API. Will push a commit.

Reply:
Pushed a commit where event domain is now optional, set via `eventEmitterProvider.eventEmitterBuilder("scope-name").setEventDomain("event-domain").build()`.

If unset, the `event.domain` attribute defaults to `unknown`.